src/main/java/com/impactupgrade/nucleus/model/ManageDonationEvent.java

```diff
@@ -31,13 +32,17 @@ public ManageDonationEvent(Environment.RequestEnvironment requestEnv) {
 
   public ManageDonationEvent(Environment.RequestEnvironment requestEnv, ManageDonationFormData formData) throws ParseException {
     this.requestEnv = requestEnv;
-    this.setDonationId(formData.recurringDonationId);
+
+    if (formData.recurringDonationId.isPresent() && !Strings.isNullOrEmpty(formData.recurringDonationId.get())) this.donationId = formData.recurringDonationId.get();
+    if (formData.recurringDonationName.isPresent() && !Strings.isNullOrEmpty(formData.recurringDonationName.get())) this.donationName = formData.recurringDonationName.get();
+
     formData.stripeToken.ifPresent(s -> this.stripeToken = s);
 
-    if (!Strings.isNullOrEmpty(formData.amount.get())) this.amount = Double.parseDouble(formData.amount.get());
-    if (formData.pauseDonationUntilDate.isPresent()) this.pauseDonationUntilDate = Utils.getCalendarFromDateString(formData.pauseDonationUntilDate.get());
-    if (formData.resumeDonationOnDate.isPresent()) this.resumeDonationOnDate = Utils.getCalendarFromDateString(formData.resumeDonationOnDate.get());
-    if (!Strings.isNullOrEmpty(formData.nextPaymentDate.get())) this.nextPaymentDate = Utils.getCalendarFromDateString(formData.nextPaymentDate.get());
+    if (formData.amount != null && formData.amount.isPresent()) this.amount = formData.amount.get();
```

Comment:
Could these ever actually be null?

Reply:
IIRC when amount is not provided by the user, it arrives to this point as a null/empty value. I had expected it to arrive as an Optional.empty(). Will have to test some more to see if this is still the case.